src/Storages/VirtualColumnUtils.cpp

```diff
@@ -115,7 +119,19 @@ NameSet getVirtualNamesForFileLikeStorage()
     return {"_path", "_file", "_size", "_time"};
 }
 
-VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription & storage_columns)
+std::map<std::string, std::string> parseHivePartitioningKeysAndValues(const std::string& path)
```

Comment:
Is there a reason to use `std::map` instead of `std::unordered_map`? Do we need keys to be sorted?